### ISSUE TITLE
Explain producers & consumers

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7424]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#53|
|Type|enhancement|
|Created|2003-07-12 20:54:34Z|

<details><summary>Searchable metadata</summary>

```
trac-id__53 53
type__enhancement enhancement
reporter__radix radix
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1058043274000000 1058043274000000
changetime__1150848222000000 1150848222000000
version__ 
owner__ 
cc__radix cc__spiv cc__cce
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
There should be a HOWTO or something somewhere which
explains registerProducer, startProducing,
pauseProducing, resumeProducing, stopProducing,
resumeConsuming, etc...

I'm not sure where it would go. The docstrings on
abstract.FileDescriptor are actually fairly good
already....

But I have in the past wondered about them, and didn't
find the abstract.FileDescriptor docs.

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="cce's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|cce commented|
|-|-|

```
#!html
<pre>
this has been explained to me a few times, so I should be able 
to create a HOWTO for producer/consumer, it is also related to 
flow since that is sort of what flow does, unfortunately, I'm very 
busy till Sept 1st, but if no one has tackled it by then, ...

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Are you going to be able to look at this?

If you can provide examples, I can integrate them into a doc (remembering that
examples need to be copyright-assigned to go into the CVS repository).
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

I'll do this next weekend.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16181]) Producer/Consumer developer document

Fixes [#7424](https://github.com/twisted/twisted/issues/7424)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
